### ISSUE TITLE
Un code à l'ancienne en + du QR pour pouvoir générer une attestation pour ma grand-mère (et des gens sans smartphones) avec mon terminal ?

### ISSUE THREAD

**Issue by hugobiwan** [LAB-MI/attestation-deplacement-derogatoire-q4-2020#143]
Bonjour !
Voilà. Mes collègues font de l'urgence "numérique" à Rennes, et ce n'est pas du tout évident de trouver des lieux qui restent ouverts pour distribuer ou imprimer des attestations derogatoires  pour des personnes déconnectées.
Et les imprimantes sont dans des lieux publics fermés actuellement, la plupart du temps (bibliotheques...).

Vu que la [version web (super bien) ](https://media.interieur.gouv.fr/deplacement-covid-19/) génère un pdf avec un QR code linké, si je veux faire une attestation pour une personne qui n'a pas de smartphone et qu'il n'y a pas d'imprimante, elle ne peut pas avoir le document sur elle. Pour montrer une version papier ou le QR code.

Par contre,  je me disais que ça serait super d'afficher sous le QR code un code aphanumérique court unique à durée de vie limitée correspondant au pdf dans le cloud (genre url shkrinkee).

Ainsi, je rempli pour mamie le formulaire web d'attestation, le site m'affiche pdf et QR code **et un code unique dessou**s. **je recopie ce code sur papier libre avec un crayon**. Elle prend ce code et ses papiers, et en cas de contrôle, les FDO **ont le choix de taper le code ou de scanner le QR pour checker le pdf dans le cloud**. 

Ceci nous libérerait de toute contrainte imprimante à côté pour générer une attestation en dépannage de quelqu'un qui n'a pas de smartphone ni ordi, notamment dans les Espaces Sociaux Communs, pour les personnes des assos solidaires etc. Et de moins courir après des imprimantes pour des publics en difficulté :-D

Bravo pour le super boulot effectué.

@hugobiwan - hugobiwan.zolnir(chez)gmail

**Comment by a2br** [comment]
Bonne idée, mais ça revient à envoyer les données dans le cloud, et ça, peu de gens apprécieront. Et puis, de toute manière -- tu sais bien que tout le monde sait dessiner des QR codes (c'est juste du binaire, après tout) ;)

**Comment by hugobiwan** [comment]
Merci pour ta réponse. Mais quand les FDO te contrôlent, j'imagine qu'elles scannent le QR code et qu'elles voient le pdf généré. Donc sans conservation de données, ll y a le document dans le cloud, pour vérification. Tout ce que je demande c'est la possibiité de se trimballer avec un numéro ou code que les FDO tapent dans leur interface de verif pour accéder au même document.
Est-ce envisageable. Au terrain ceci nous aiderait beaucoup.

**Comment by maximebochon** [comment]
@hugobiwan Justement le système du QR-Code n'est pas un lien vers une ressource stockée sur internet, mais bien un usage standard de codage d'informations textuelles sous forme de code-barre bidimensionnel.

Dit plus simplement : si vous scannez vous-même votre QR-Code vous verrez son contenu, et vous constaterez qu'il s'agit des informations saisies, accompagnées d'un horodatage de création, le tout sous forme de texte brut (sans accent pour des raisons d'encodage), et aucunement d'un lien vers le PDF.

De plus le PDF est généré par l'appareil de l'utilisateur depuis son navigateur, justement dans le but d'éviter tout problème lié à la conservation d'informations personnelles. L'inverse entraînerait certainement une levée de bouclier d'une partie de la population ou d'associations de protection, nécessiterait une déclaration à la CNIL, ainsi probablement que la mise en œuvre du RGPD.

J'ai le sentiment que votre demande telle qu'elle est présentée a peu de chances d'aboutir mais qu'elle correspond a une réelle problématique qui demande une solution pratique. Il faudrait peut-être regarder le problème sous un autre angle pour trouver une solution sans a priori sur la manière de faire.

**Comment by maximebochon** [comment]
@hugobiwan Je viens de relire dans le détail votre problématique.

Dans l'absolu, ce que vous demandez requiert que la personne possède un papier et un crayon, et soit en mesure d'écrire sans aucune erreur un identifiant unique potentiellement indigeste.

Hors je me dis qu'une telle personne (ou son tuteur) pourra probablement être en mesure d'écrire une attestation simplifiée du type :
```
<Prénom> <NOM>, <date de naissance>, <lieu de naissance>
<adresse>, <code postal>, <ville>
<date de sortie> à <heure de sortie>
<motif simplifié de sortie> (travail, achats, etc.)
<signature>
```

La problématique reviendrait alors à ce que cette attestation papier simplifiée, plus facile à produire que l'attestation complète, soit valide juridiquement, validité éventuellement soumise à la condition de posséder une carte handicapée ou une attestation personnelle délivrée par un organisme d'aide aux personnes en difficulté (je ne connais pas du tout le contexte pratique dont vous faites l'écho ici, j'espère que vous excuserez mes maladresses).

**Comment by hugobiwan** [comment]
Merci beaucoup @maximebochon !
Donc le data est directement contenu dans le QR code, et le pdf est généré en local. 
C'est super, en terme de confiance numérique !
Ensuite, une attestation à main levée à valeur légale en cas de contrôle ? 
Ca fait deux bonnes infos si vous pouvez me confirmer le point 2.

**Comment by maximebochon** [comment]
Le premier point est tout à fait avéré dans la version actuelle du code. C'est effectivement très bien pour la confiance numérique. Tant qu'on y est, sachez que le code source de _TousAntiCovid_ est ouvert et disponible ici : https://gitlab.inria.fr/stopcovid19

Concernant le second point, je vais vous décevoir : je ne suis qu'un simple citoyen qui réagit à votre problématiquement avec les informations à ma disposition. En l'occurrence j'ai formulé une hypothèse, d'où l'emploi du conditionnel, et je n'ai aucun pouvoir pour faire changer les choses, ni concernant le générateur d'attestation, ni concernant le point juridique évoqué. Je suggère juste une idée de solution, à faire remonter à je ne sais qui ayant un réel pouvoir de décision. La suite vous appartient ;-)

**Comment by a2br** [comment]
> Ensuite, une attestation à main levée à valeur légale en cas de contrôle ?

Oui, tout à fait, du moment que tous les champs de la version papier sont retranscrits, une attestation est valide même rédigée à la main.

**Comment by hugobiwan** [comment]
Je vous remercie beaucoup @maximebochon @a2br . Je comprends tres bien qu'on ne puisse embarquer/noter sur un papier un code raccourci en texte sans smartphone, relié à un doc pdf, à partir du moment où le pdf n'est que dans le client. Si quelqu'un avait une idée pour les personnes sans smartphone ni imprimante qui pourraient être aidées par des possesseurs de smartphone, je reste preneur !
